pkg/controller/statusmanager/pod_status.go

```diff
@@ -72,37 +73,74 @@ func FindStatusCondition(conditions []corev1.NodeCondition, conditionType corev1
 	return nil
 }
 
-func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.NodeCondition) {
+func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.NodeCondition) (changed bool) {
+	changed = false
 	if conditions == nil {
 		conditions = &[]corev1.NodeCondition{}
 	}
 	existingCondition := FindStatusCondition(*conditions, newCondition.Type)
 	if existingCondition == nil {
 		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
 		*conditions = append(*conditions, newCondition)
-		return
+		return true
 	}
 
 	if existingCondition.Status != newCondition.Status {
 		existingCondition.Status = newCondition.Status
 		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		changed = true
 	}
 
 	existingCondition.Reason = newCondition.Reason
 	existingCondition.Message = newCondition.Message
+	return changed
 }
 
-func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, reason string, message string) {
+func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
+	if sharedInfo.LastNetworkAvailable == nil {
+		sharedInfo.LastNetworkAvailable = make(map[string]time.Time)
+	}
 	if ready == true {
 		log.Info(fmt.Sprintf("Setting status NetworkUnavailable to false for node %s", nodeName))
 	} else {
 		log.Info(fmt.Sprintf("Setting status NetworkUnavailable to true for node %s", nodeName))
+		delete(sharedInfo.LastNetworkAvailable, nodeName)
+	}
```

Comment:
Line 103-108 and line 114-117 can put together.

Reply:
done